Weitere A11y Enhancements für das Navigation Bundle (Case 143343)

megatronCGN opened the pull request.

megatronCGN committed:
Setze Aria-Label mit Default-Werten auf Nav-Elemente

polarbirke commented on src/Resources/views/Navigation/breadcrumbs-blocks-BEM.html.twig:
```diff
@@ -6,7 +6,7 @@
 {% block breadcrumbs %}
     {% if breadcrumbs is defined and breadcrumbs|length > 1 %}
-        <nav class="{{ block('prefix_final') ~ 'breadcrumbs' }} {% block breadcrumbs_class %}{% endblock %}">
+        <nav aria-label="{{ breadcrumbs_landmark|default('Breadcrumbs') }}" class="{{ block('prefix_final') ~ 'breadcrumbs' }} {% block breadcrumbs_class %}{% endblock %}">
```
Vorschlag: "Brotkrumennavigation", so nennt es auch www.bundestag.de, und es vermeidet ein englisches Wort, das auf einer `lang="de"` Seite mit der falschen Aussprache vorgelesen würde.

megatronCGN replied:
ja, finde ich gut

megatronCGN committed:
Update breadcrumbs-blocks-BEM.html.twig

Deutsche Bezeichnung für 'Breadcrumbs'

polarbirke commented on src/Resources/views/Navigation/navigation-blocks-BEM.html.twig:
```diff
@@ -15,7 +15,7 @@
         {% if visibleNodes is not empty %}
             {% if level == 0  %}
-                <nav class="{{ navigation_base_class }} {% block navigation_additional_class %}{% endblock %}">
+                <nav aria-label="{{ navigation_landmark|default('Hauptnavigation') }}" class="{{ navigation_base_class }} {% block navigation_additional_class %}{% endblock %}">
```
Ist das hier eine gute Idee? Würde man _dieses_ Partial via `extend`  o.ä. für alle möglichen Navigationen verwenden?

Ich frage mich, ob es hier nicht besser wäre, eine Exception zu werfen, wenn bei der Verwendung kein Name übergeben wird. Das trainiert uns, und es kann nicht passieren, dass Haupt-, Sub- und Footernavigation alle "Hauptnavigation" heißen.

megatronCGN replied:
Good Point. D.h. hier einfach den Default Filter entfernen right?

megatronCGN replied:
Das wäre allerdings ein BC.

polarbirke replied:
Hm, ja. Aber noch überschaubar und man könnte die 4-5 Projekte direkt fixen? Weiß nicht. Bei internen Paketen habe ich manchmal die Tendenz, es mit Semver nicht zu übertreiben, aber eine 5.0.0 kostet auch nix.

~Leichte~ Mittelstarke Tendenz zu 5.0.0 aber 🤷‍♀️ 

Auf >= 4.1 sind bisher:
<img width="370" alt="Bildschirmfoto 2022-03-24 um 14 56 47" src="https://user-images.githubusercontent.com/3763421/159932367-c96c20e8-a783-43f6-9274-f415a5b77a6e.png">

megatronCGN replied:
Ich tendiere auch zu 5.x obwohl es so eine kleine Änderung ist.

megatronCGN replied:
Alternativ könnten wir uns in der Ausgabe auch mit `{% if navigation_landmark is not defined %}Hast du nicht was vergessen?!{% endif %}` anschreien.

megatronCGN replied:
<img width="677" alt="Bildschirmfoto 2022-03-24 um 15 30 27" src="https://user-images.githubusercontent.com/7603184/159939155-7a69782c-8f82-4119-aa07-fac3e73bebc4.png">

megatronCGN replied:
Ich tendiere dazu, mich anschreien zu lassen.

polarbirke replied:
Wo siehst du den Vorteil vom "visuellen Anschreien" gegenüber einer Exception? Ich habe das anfangs auch gern so gemacht, aber finde die Exception genauso aussagekräftig und noch, äh, lauter 😄

megatronCGN replied:
Kein richtiger Vorteil. Es wäre halt nur kein major update sondern ein feature. Was für mich mehr der Natur der Änderung entspräche.

polarbirke replied:
…aber wäre es nicht immer noch ein BC, weil du sonst mit "nur" dem Paketupdate + Depployment auf der Live-Seite die schönen Meldungen sehen würdest? Steh ich auf dem Schlauch?

megatronCGN replied:
Nunja, du könntest immerhin noch navigieren. :D

megatronCGN replied:
Überzeugt. Ich mache eine 5er.

megatronCGN committed:
Update navigation-blocks-BEM.html.twig

Entferne Default für `navigation_landmark`, damit bei mehreren Navigation, die das Template extenden, nicht alle dieselbe Landmark haben. Wir zwingen uns dadurch darüber nachzudenken und sinnige Landmarks zu vergeben.

megatronCGN committed:
Update navigation-blocks-BEM.html.twig

Entferne den default-Filter

megatronCGN committed:
Update CHANGELOG.md

Füge Hinweis für Version 5 hinzu.

megatronCGN committed:
Update CHANGELOG.md

Korrigiere Attribut Name

megatronCGN committed:
Update CHANGELOG.md

Verbessere Beschreibung
